src/uwtools/tests/drivers/test_jedi.py

```diff
@@ -0,0 +1,165 @@
+"""
+JEDI driver tests.
+"""
+import datetime as dt
+import logging
+from unittest.mock import DEFAULT as D
+from unittest.mock import Mock, patch
+
+import pytest
+import yaml
+from pytest import fixture
+
+from uwtools.drivers import jedi
+from uwtools.scheduler import Slurm
+from uwtools.tests.support import fixture_path, regex_logged
+
+# Fixtures
+
+
+@fixture
+def cycle():
+    return dt.datetime(2024, 2, 1, 18)
+
+
+# Driver fixtures
+
+
+@fixture
+def config(tmp_path):
+    return {
+        "jedi": {
+            "execution": {
+                "batchargs": {
+                    "export": "NONE",
+                    "nodes": 1,
+                    "stdout": "/path/to/file",
+                    "walltime": "00:02:00",
+                },
+                "envcmds": [
+                    "cmd1",
+                    "cmd2",
+                    "module load right-modules",
+                    "module load jedi-modules",
+                ],
+                "executable": "/scratch2/BMC/zrtrr/Naureen.Bharwani/build/bin/qg_forecast.x",
+                "mpiargs": ["--export=ALL", "--ntasks $SLURM_CPUS_ON_NODE"],
+                "mpicmd": "srun",
+            },
+            "configuration_file": {
+                "base_file": str(fixture_path("jedi.yaml")),
+                "update_values": {"jedi": {}},
+            },
+            "run_dir": str(tmp_path),
+        },
+        "platform": {
+            "account": "me",
+            "scheduler": "slurm",
+        },
+    }
+
+
+@fixture
+def config_file(config, tmp_path):
+    path = tmp_path / "config.yaml"
+    with open(path, "w", encoding="utf-8") as f:
+        yaml.dump(config, f)
+    return path
+
+
+@fixture
+def driverobj(config_file, cycle):
+    return jedi.JEDI(config=config_file, cycle=cycle, batch=True)
+
+
+# Driver tests
+
+
+def test_JEDI(driverobj):
+    assert isinstance(driverobj, jedi.JEDI)
+
+
+def test_JEDI_dry_run(config_file, cycle):
+    with patch.object(jedi, "dryrun") as dryrun:
+        driverobj = jedi.JEDI(config=config_file, cycle=cycle, batch=True, dry_run=True)
+    assert driverobj._dry_run is True
+    dryrun.assert_called_once_with()
+
+
+@pytest.mark.parametrize("key,task,test", [("files_to_copy", "files_copied", "is_file")])
+def test_JEDI_files_copied(caplog, config, cycle, key, task, test, tmp_path):
+    logging.getLogger().setLevel(logging.INFO)
+    with patch.object(jedi, "is_file") as is_file:
+        pass
+
+
+@pytest.mark.parametrize("key,task,test", [("files_to_link", "files_linked", "is_symlink")])
+def test_JEDI_files_copied(caplog, config, cycle, key, task, test, tmp_path):
+    logging.getLogger().setLevel(logging.INFO)
+    # with patch.object(jedi, "is_symlink") as is_symlink:
+
+
+def test_JEDI_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        files_copied=D,
+        files_linked=D,
+        runscript=D,
```

Comment:
Does `validate_only` need to be apart of this provisioned run dir. Should this be an optional argument if supplie dby the user?

i.e. 
```
if validate_only supplied:
    add validate_only in mocks

```

Reply:
It could be a nice way of ensuring that the files we're generating for configuration are consistent with the executable before wasting compute resources.

Reply:
I made some related comments above about where and when the `validate_only` task should run.